src/x86/bmi2.rs

```diff
@@ -16,7 +16,8 @@ use assert_instr::assert_instr;
 /// the low half and the high half of the result.
 #[inline(always)]
 // LLVM BUG (should be mulxl): https://bugs.llvm.org/show_bug.cgi?id=34232
-#[cfg_attr(test, assert_instr(imul))]
+#[cfg_attr(all(test, target_arch = "x86_64"), assert_instr(imul))]
+#[cfg_attr(all(test, target_arch = "x86"), assert_instr(mulx))]
```

Comment:
what? does this generate the right instruction  (`mulx`) on x87 but not x86_64 ? If so the LLVM bug is incorrect, because there the state of the art is that LLVM never uses `mulx` on 32bit integers. It is not profitable.

Reply:
Yeah I dunno :(

Reply:
I can't reproduce this with clang `-m32`, weird :/
https://godbolt.org/g/EVPwHw

Reply:
Maybe differing versions of LLVM? Or maybe we're doing the wrong target features? You may want to try checking this branch out and fiddling with the attribute as that'll show the disassembly in the test failure